### PR TITLE
XyChart: Support adding calcs to legend

### LINKED ISSUE #85034: XYChart2: Some legend options don't work
visibility off, calc values

image

### PR DESCRIPTION
**Pull request by adela-almasan**
legend_calcs

Fixes #85034

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/plugins/panel/xychart/v2/XYChartPanel.tsx [hunk 1]
```diff
@@ -68,17 +78,82 @@ export const XYChartPanel2 = (props: Props2) => {
           getItemKey: () => `${idx}-${s.name.value}`,
           fieldName: yField.state?.displayName ?? yField.name,
           disabled: yField.state?.hideFrom?.viz ?? false,
+          getDisplayValues: () => {
```
Comment (baldm0mma) [inline public/app/plugins/panel/xychart/v2/XYChartPanel.tsx:81]: Is this function different than the one used in PlotLegend?

Comment (baldm0mma) [reply public/app/plugins/panel/xychart/v2/XYChartPanel.tsx:81]: Or in the original XYChart? Maybe we can DRY it out if it is?

Comment (adela-almasan) [reply public/app/plugins/panel/xychart/v2/XYChartPanel.tsx]: Pushed a quick update. The plan was to extract it to a separate component, but not today apparently 😅

**Review by baldm0mma — APPROVED** [review]
Nice!

**Pull request merged**